app/src/main/java/org/thoughtcrime/securesms/database/ApnDatabase.java

```diff
@@ -87,7 +87,7 @@ private ApnDatabase(final Context context) throws IOException {
       throw new IOException("couldn't make databases directory");
     }
 
-    Util.copy(context.getAssets().open(ASSET_PATH, AssetManager.ACCESS_STREAMING),
+    StreamUtil.copy(context.getAssets().open(ASSET_PATH, AssetManager.ACCESS_STREAMING),
               new FileOutputStream(dbFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileOutputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileOutputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).